services/spar/src/Spar/App.hs

```diff
@@ -208,12 +211,28 @@ autoprovisionSamlUserWithId buid suid mbName managedBy = do
   let teamid = idp ^. idpExtraInfo . wiTeam
   scimtoks <- wrapMonadClient $ Data.getScimTokens teamid
   if null scimtoks
-    then createSamlUserWithId buid suid mbName managedBy
+    then do
+      createSamlUserWithId buid suid mbName managedBy
+      validateEmailIfExists buid suid
```

Comment:
renamed to `ifNeeded` maybe?

Reply:
I'd be ok with `ifAvailable`, or `ifPresent`.  It's not about the need, it's about the presence.